syntax = "proto3";

service DBNode {
    rpc Count(RangeRequest) returns (SingleResponse) {}

    rpc Sum(RangeRequest) returns (SingleResponse) {}

    rpc Insert(stream DataRequest) returns (InsertDataAcknowledgement) {}
}

message RangeRequest {
    int32 idStart = 1;
    int32 idEnd = 2;
}

message SingleResponse {
    int32 result = 1;
}

message DataRequest {
    int32 key = 1;
    int32 value = 2;
}

message InsertDataAcknowledgement {
}